### PR TITLE
Implement handler "interface" and add function for mapping handlers to IRC

### PR DESCRIPTION
**Pull request by Zedjones**
This adds a type alias for IRC handlers in a `handlers.go` file within `internal/handlers/irc` and adds a function within that returns a map of strings (girc event types) to handler functions. This will be edited as handlers are created, and then the handlers get added to the client as such
```Go
for eventType, handler := range c.HandlerMap {
    c.Handlers.Add(eventType, handler(c))
}
```

### CODE REVIEW

File: cmd/teleirc.go [hunk 1]
```diff
@@ -4,16 +4,19 @@ package main
...
+	tgbotapi "github.com/go-telegram-bot-api/telegram-bot-api"
 	"github.com/ritlug/teleirc/internal"
 	"github.com/ritlug/teleirc/internal/handlers/irc"
 	tg "github.com/ritlug/teleirc/internal/handlers/telegram"
 )
 
-var (
+const (
 	version = "v2.0"
+)
```
Comment (jwflory) [inline cmd/teleirc.go]: Good call :+1:

File: cmd/teleirc.go [hunk 2]
```diff
@@ -34,8 +37,8 @@ func main() {
 
 	settings, err := internal.LoadConfig(*flagPath)
 	if err != nil {
-		fmt.Println(err)
-		return
+		fmt.Fprintln(os.Stderr, err)
+		os.Exit(1)
```
Comment (jwflory) [inline cmd/teleirc.go]: Aha – is this the exit code we talked about, in the context of systemd service files?

Comment (Zedjones) [reply cmd/teleirc.go]: Yep! Eventually we'll change it to use a proper logger, but this is better than nothing for the moment.

File: internal/handlers/irc/irc.go [hunk 3]
```diff
@@ -64,20 +70,7 @@ that were passed in to NewClient
...
-}
-
-/*
-connectHandler return a function to use as the connect handler for girc,
-so that the specified channel is joined after the server connection is established
-*/
-func connectHandler(c Client) func(*girc.Client, girc.Event) {
-	return func(gc *girc.Client, e girc.Event) {
-		c.Cmd.Join(c.Settings.Channel)
+	for eventType, handler := range getHandlerMapping() {
```
Comment (Tjzabel) [inline internal/handlers/irc/irc.go]: nice way of doing this :smile:

Comment (Tjzabel) [inline internal/handlers/irc/irc.go]: Nice way of doing this :smile:

File: internal/handlers/irc/handlers.go [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+			c.SendMessage(formatted)
+		}
+	}
+}
+
+/*
+getHandlerMapping returns a mapping of girc event types to handlers
+*/
+func getHandlerMapping() map[string]Handler {
+	return map[string]Handler{
```
Comment (Tjzabel) [inline internal/handlers/irc/handlers.go]: ohhh, fancy

File: internal/handlers/irc/handlers.go [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+/*
+Handler specifies a function that handles an IRC event
+In this case, we take an IRC client and return a function that
+handles an IRC event
+*/
+type Handler = func(c Client) func(*girc.Client, girc.Event)
+
+/*
+connectHandler return a function to use as the connect handler for girc,
```
Comment (Tjzabel) [inline internal/handlers/irc/handlers.go]: super pedantic, but "connectHandler return**s** a function..."

File: .gitignore [hunk 6]
```diff
@@ -30,3 +30,7 @@ pids
 
 # Running config file! Don't commit this unless you know what you're doing.
 .env
+
+# Generated executables
+teleirc
+cmd/cmd
```
Comment (Tjzabel) [inline .gitignore]: For some reason VS Code doesn't put EOF/newlines at the end of the file, so GitHub adds a symbol to let us know. Not sure if this hurts anything?

File: internal/handlers/irc/irc.go (hunk 3 above)
Comment (Zedjones) [reply internal/handlers/irc/irc.go]: Thank you! 😄

File: .gitignore (hunk 6 above)
Comment (Zedjones) [reply .gitignore]: It shouldn't hurt anything, it's just a stylistic preference I think? Maybe it makes a difference on Windows? Not sure

**Comment by Zedjones** [comment]
Gonna go ahead and merge this, thanks for the reviews :+1:

**Pull request merged**